### ISSUE TITLE
Regress tests need large tx tests

### ISSUE THREAD

**Issue by ringerc** [2ndQuadrant/bdr#57]
Per #54 - while it's unclear what's going on there and we generally handle big tx's fine, we need to make that part of our automated testing framework.

**Comment by ringerc** [comment]
@kforte90 Have you had a look at the regression tests? I'd be interested in getting you adding to the test infrastructure and test automation directly where possible so we can avoid running manual tests and so that tests and test failures can be more easily reproduced.

**Comment by kforte90** [comment]
Yes, I've looked at the regression tests.  I'm working on some automated multi-user or multi-transaction tests.   There are two flavors of automated multi-user transaction tests: (1)  deterministic interleaved transactions tests where multiple nodes 'agree' on the correct result and (2) multiple long and short transactions that bombard the cluster for a measured period of time.
What I've seen is the following and it seems incorrect to me but maybe the behavior is expected from a async replication manager:
```
create table t1 (c1 int, primary key (c1));
insert into t1 values (1);
--trx1, node 1
begin;
update t1 set c1 = 2;
     ---trx2, node 2
    begin;
    update t1 set c1 = 3;
         --back to trx1, node 1:
           commit;
                  --back to trx2, node 2:
                    commit;
--------------------
--Now check that both nodes see the last committed row - should see c1 = 3
trx1, node 1:
select * from t1;
 c1 
----
  2
(1 row)
--trx2, node 2:
 select * from t1;
 c1 
----
  3
(1 row)
```
I would have thought that with the 'last commit wins' , both nodes would see '3' as the value for c1 in table t1 with 'track_commit_timestamp = on'?  
I'll assign the issue of creating tests to me. I do need to understand if this is the correct behavior or not.

**Comment by kforte90** [comment]
Another question: are 'large' transactions the same as 'long' transactions? Or, do you mean 'large' in terms of the data size. I haven't heard the term 'large' transaction before.

**Comment by ringerc** [comment]
Both, really, but I'm interested mainly in tx's that affect many rows,
and/or many tables.

Duration is significant too, but a different matter really.

On 14 April 2015 at 05:11, kforte90 <notifications@github.com> wrote:

> Another question: are 'large' transactions the same as 'long'
> transactions? Or, do you mean 'large' in terms of the data size. I haven't
> heard the term 'large' transaction before.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/2ndQuadrant/bdr/issues/57#issuecomment-92499956>.
>



-- 
 Craig Ringer                   http://www.2ndQuadrant.com/
 PostgreSQL Development, 24x7 Support, Training & Services
